gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -323,6 +324,14 @@ public void launchJob(JobListener jobListener) throws JobException {
     }
   }
 
+  /**
+   * Subclasses can override this method to do whatever processing on the taskStates,
+   * e.g., aggregate task-level metrics into job-level metrics.
+   */
+  protected void processTaskStates(List<TaskState> taskStates) {
```

Comment:
`postProcessTaskStates` may be a better name? BTW: I will make big changes to the commit logic here to work on a per-dataset basis. I would suggest you move the method out of the `if` block.